File: CORREDOR/Tipos Constructores/JS/TelefonoInteligente.js
const Electronico = require("./Electronico");

//Constructor Sobrecargado
class TelefonoInteligente extends Electronico {
    constructor(modelo, almacenamiento) {
        super(); // Llama al constructor sin parámetros de Electronico
        this.modelo = modelo;
        this.almacenamiento = almacenamiento;
    }

    encender() {
        console.log("Encendiendo el teléfono...");
    }

    informacion() {
        console.log("Marca:", this.marca);
        console.log("Año:", this.año);
        console.log("Modelo:", this.modelo);
        console.log("Almacenamiento:", this.almacenamiento + " GB");
    }
}
module.exports = TelefonoInteligente;

const t1 = new TelefonoInteligente("Galaxy A54", 128);
t1.encender();
t1.informacion();

File: CORREDOR/Tipos Constructores/Java/TelefonoInteligente.java
package Java;

public class TelefonoInteligente extends Electronico implements Dispositivo{
    String modelo;
    int almacenamiento;

    // Constructor sobrecargado
    public TelefonoInteligente(String modelo, int almacenamiento) {
        super();// Llama al constructor sin parámetros de Electronico
        this.modelo = modelo;
        this.almacenamiento = almacenamiento;
    }

    @Override
    public void encender() {
        System.out.println("Encendiendo el teléfono...");
    }

    @Override
    void informacion() {
        System.out.println("Marca: " + marca);
        System.out.println("Año: " + año);
        System.out.println("Modelo: " + modelo);
        System.out.println("Almacenamiento: " + almacenamiento + " GB");
    }
}